### PR TITLE
docs: Add documentation for obs_frontend_add_dock

### PR DESCRIPTION
**Pull request by Xaymar**
### Description
Adds documentation for the `obs_frontend_add_dock` function.

### Motivation and Context
Wanted to create docks, and initially hit a brick wall because the function wasn't documented well enough.

### How Has This Been Tested?
Created a dock, and successfully restored it's state.

### Types of changes

- Documentation (a change to documentation pages)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: docs/sphinx/reference-frontend-api.rst [hunk 1]
```diff
@@ -623,3 +623,24 @@ Functions
 
    :return: The value of the position of the T-bar to, with a value in 0-1023.
    :rtype: int
+
+---------------------------------------
+
+.. function:: void* obs_frontend_add_dock(void* dock_widget)
```
Comment (RytoEX) [inline docs/sphinx/reference-frontend-api.rst:629]:
```
void *obs_frontend_add_dock(void *dock_widget)
```

File: docs/sphinx/reference-frontend-api.rst [hunk 2]
```diff
@@ -147,10 +147,10 @@ Structures/Enumerations
    - **OBS_FRONTEND_EVENT_VIRTUALCAM_STOPPED**
 
      Triggered when the virtual camera is stopped.
-   
+
    - **OBS_FRONTEND_EVENT_TBAR_VALUE_CHANGED**
 
-     Triggered when the transition bar is moved. 
+     Triggered when the transition bar is moved.
```
Comment (RytoEX) [inline docs/sphinx/reference-frontend-api.rst:153]: Unrelated whitespace changes.

File: docs/sphinx/reference-frontend-api.rst [hunk 3]
```diff
@@ -605,7 +605,7 @@ Functions
 .. function:: void obs_frontend_release_tbar(void);
 
    Emulate a mouse button release on the transition bar and determine transition status.
-
```
Comment (RytoEX) [inline docs/sphinx/reference-frontend-api.rst:608]: Unrelated whitespace changes.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: docs/sphinx/reference-frontend-api.rst (hunk 2 above)
Comment (Xaymar) [reply docs/sphinx/reference-frontend-api.rst]: Will fix. This was automatically added by VSCode as I've set it up to format the documents on save.

File: docs/sphinx/reference-frontend-api.rst (hunk 3 above)
Comment (Xaymar) [reply docs/sphinx/reference-frontend-api.rst]: Will fix. This was automatically added by VSCode as I've set it up to format the documents on save.

File: docs/sphinx/reference-frontend-api.rst (hunk 1 above)
Comment (Xaymar) [reply docs/sphinx/reference-frontend-api.rst]: Fixed

**Comment by Xaymar** [comment]
Closing as requested.
